ztp/extra-manifests-builder/05-kdump-config/kdump-remove-ice-module.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+# This script removes the ice module from kdump to prevent kdump failures on certain servers.
+# This is a temporary workaround for RHELPLAN-138236 and can be removed when that issue is
+# fixed.
+
+set -x
+
+SED="/usr/bin/sed"
+GREP="/usr/bin/grep"
+
+# override for testing purposes
+KDUMP_CONF="${1:-/etc/sysconfig/kdump}"
+REMOVE_ICE_STR="module_blacklist=ice"
+
+# exit if file doesn't exist
+[ ! -f ${KDUMP_CONF} ] && exit 0
+
+# exit if file already updated
+${GREP} -Fq ${REMOVE_ICE_STR} ${KDUMP_CONF} && exit 0
+
+# Target line looks something like this:
+# KDUMP_COMMANDLINE_APPEND="irqpoll nr_cpus=1 ... hest_disable"
+# Use sed to match everything between the quotes and append the REMOVE_ICE_STR to it
+${SED} -i 's/^KDUMP_COMMANDLINE_APPEND="[^"]*/& '${REMOVE_ICE_STR}'/' ${KDUMP_CONF} || exit 0
```

Comment:
Is there a reason we are squashing sed's exit code here? Wouldn't we want an editing error to be reported/recorded?

Reply:
My thinking (could be wrong) was that this is a non-essential change and that I didn't want to cause the machineconfig to fail any case. I didn't want to affect the health of the system in any way, regardless of what happens in this machineconfig.